homeassistant/components/sensor/nextbus.py

```diff
@@ -0,0 +1,271 @@
+"""
+NextBus sensor.
+
+This sensor displays minutes until the next bus/train via the NextBus API.
+
+More information on this service can be found on https://www.nextbus.com/
+
+Configuration sample:
+
+    sensors:
+      - platform: nextbus
+          agency: sf-muni  # agency tag from NextBus
+          route: F         # route tag from NextBus
+          stop: 35184      # stop tag from NextBus
+          name: 'My stop'  # optional name to use for the sensor
+
+It is possible to get the tag information from the NextBus website.
+
+    1. Visit https://www.nextbus.com/
+    2. Use the drop downs to select the transit system, route, direction, and
+        stop
+    3. Extract the tags from the URL. It is constructed with the following
+        pattern:
+            https://www.nextbus.com/#!/<agency>/<route>/<direction>/<stop>
+
+If tags are incorrect, valid ones will be displayed in the logs as a
+convenience.
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+
+
+REQUIREMENTS = ['py_nextbus==0.1.2']
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONF_AGENCY = 'agency'
+CONF_ROUTE = 'route'
+CONF_STOP = 'stop'
+
+
+ICON = 'mdi:bus'
+ATTR_MINUTES = 'min'
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_AGENCY): cv.string,
+    vol.Required(CONF_ROUTE): cv.string,
+    vol.Required(CONF_STOP): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+
+def get_predictions(stops, agency):
+    """Return predictions for provided stops."""
+    from py_nextbus import NextBusClient  # pylint: disable=import-error
```

Comment:
Pylint says it's unable to import. I believe this is because the module isn't installed in the virtualenv when I run `pylint` directly, but I think it actually will pass with `tox` as dependencies are installed. Removing.